frontend/src/queries/nodes/DataTable/renderColumn.tsx

```diff
@@ -31,8 +31,23 @@ export function renderColumn(
     } else if (isHogQLQuery(query.source)) {
         if (typeof value === 'string') {
             try {
-                if ((value.startsWith('{') && value.endsWith('}')) || (value.startsWith('[') && value.endsWith(']'))) {
-                    return <ReactJson src={JSON.parse(value)} name={key} collapsed={1} />
+                if (value.startsWith('{') && value.endsWith('}')) {
+                    return (
+                        <ReactJson
+                            src={JSON.parse(value)}
+                            name={key}
+                            collapsed={Object.keys(JSON.stringify(value)).length > 10 ? 0 : 1}
+                        />
+                    )
+                }
+                if (value.startsWith('[') && value.endsWith(']')) {
+                    return (
+                        <ReactJson
+                            src={JSON.parse(value)}
+                            name={key}
+                            collapsed={JSON.stringify(value).length > 10 ? 0 : 1}
+                        />
+                    )
```

Comment:
I may be just missing something, but I don't see how this is connected to the database scene, can you explain?

Reply:
Without this, if I'd click into a few database tables (`session_recording_events`? not sure anymore), I'd get a MASSIVE list of expanded JSON * 100, which would cause my computer to freeze. This prevents that.